linked_list.py

```diff
@@ -7,12 +7,16 @@ def __init__(self, val=None, next=None):
 class list(object):
     def __init__(self, head=None):
         self.head = head
+        self.list_size = 0
+        if self.head:
+            self.list_size = 1
```

Comment:
An interesting artifact of Python is that the booleans `True` and `False` actually have a mathematical value (`1` and `0`, respectively).  You could simplify this using that knowledge:

```python
    def __init__(self, head=None):
        self.head = head
        self.list_size = 0 + bool(head)
```